5417 feeback reports stt UI

mattcoleanderson opened the pull request:
## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #5417 

This PR implements the STT Feedback Report UI, allowing STT Data Analysts to view and download their quarterly feedback reports. 

### Key changes

#### Frontend
  - New STTFeedbackReports component with fiscal year selector and paginated reports table for Data Analysts
  - New STTFeedbackReportsTable component for displaying reports with download functionality
  - New FeedbackReportAlert banner displayed on TANF Data Files page when feedback reports are available
  - Refactored FeedbackReports to conditionally render Admin vs STT views based on permissions
  - Updated Header navigation to show Feedback Reports tab for all users with view permission (not just admins)
  - New fileDownload.js utility for blob file downloads
  - New auth selectors: accountCanViewFeedbackReports (view) and accountCanUploadFeedbackReports (admin upload)

#### Backend
  - New email template and helper for feedback report availability notifications
  - Email notifications sent to approved Data Analysts when reports are created for their STT
  - Added year, quarter, and latest query params to ReportFile viewset for filtering

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose up --build
cd tdrs-backend && docker-compose up --build
```

Part 1: Upload Report as OFA Admin

  1. Sign in as an OFA Admin user at http://localhost:3000/
  2. Navigate to Feedback Reports in the header navigation
  3. Verify the page shows a fiscal year dropdown at the top with no selection
  4. Select Fiscal Year: 2025 from the dropdown
  5. Verify content appears below the horizontal rule with header "Fiscal Year 2025 — Upload Feedback Reports"
  6. Verify the date picker appears with label "Data extracted from database on" and hint text "mm/dd/yyyy"
  7. Test validation errors:
    - Click "Upload & Notify States" without selecting a file → verify error "No file selected."
    - Select a non-zip file → verify error "Invalid file. Make sure to select a zip file."
    - Select a zip file with mismatched year (e.g., FY2024_...zip when FY 2025 is selected) → verify error
  about fiscal year mismatch
    - Try to upload without selecting a date → verify error "Choose the date that the data you're uploading
  was extracted from the database."
  8. Upload the test file tdrs-backend/tdpservice/reports/test/data/FY2025_valid_single_stt.zip
  9. Select a date using the date picker (e.g., 01/15/2025)
  10. Click "Upload & Notify States"
  11. Verify success message and the upload appears in the Upload History table
  12. Verify the table shows columns: Feedback Uploaded On, Data Extracted On, Notifications Sent On, Status,
  Error, File, Downloaded by
  13. Sign out

  Part 2: View Report as Data Analyst (Matching STT)

  14. Sign in as a Data Analyst user assigned to Alabama (STT Code: 01)
  15. Navigate to Feedback Reports in the header navigation
  16. Verify you see a fiscal year dropdown with placeholder "- Select Fiscal Year -"
  17. Verify no content appears below the <hr> until you select a year
  18. Select Fiscal Year: 2025 from the dropdown
  19. Verify content appears with header: "Alabama — Fiscal Year 2025 — Feedback Reports"
  20. Verify H3 says "Feedback Reports" (NOT "Fiscal Year 2025 Feedback Reports")
  21. Verify the reports table shows columns:
    - "Generated on" (formerly "Feedback generated on")
    - "Reflects data submitted through" (formerly "Fiscal quarters included in feedback") - shows the date
  extracted value (e.g., "01/15/2025")
    - Files (download link)
  22. Click the download link and verify the file downloads successfully

  Part 3: Verify Alert on Data Files Page

  23. Navigate to Data Files in the header navigation
  24. Select the following parameters to match the uploaded report:
    - File Type: TANF (or SSP)
    - Fiscal Year: 2025
    - Quarter: Any quarter
  25. Verify an info alert banner appears stating feedback reports are available
  26. Click the X button to dismiss the alert
  27. Refresh the page and verify the alert stays dismissed for FY 2025
  28. Change to a different fiscal year (e.g., 2024) and then back to 2025 → verify alert stays dismissed
  29. (Optional) If you upload another report for FY 2025, verify the alert reappears with the new report

  Part 4: Verify STT Isolation

  30. Sign out and sign in as a Data Analyst for a different STT (NOT Alabama)
  31. Navigate to Feedback Reports → select Fiscal Year 2025
  32. Verify the reports table shows "No data available"
  33. Navigate to Data Files → select FY 2025 → verify no alert banner appears

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [ ] STT feedback report UI implemented
+ [ ] Error handling for file download implemented
+ [ ] STT UI notification on the datafiles page implemented
+ [ ] Testing Checklist has been run and all tests pass

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?


### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

mattcoleanderson committed:
Create STT Feedback Report Page

mattcoleanderson committed:
conditionally choose title and subtitle of Feedback Reports based on abilty to upload

mattcoleanderson committed:
update the Data Reporting Reference table to more closely match the styling of similar tables in TDP

mattcoleanderson committed:
add strong tag to column headers

mattcoleanderson committed:
remove unused import

mattcoleanderson committed:
fix button styling for download link

mattcoleanderson committed:
bug where search happend upon a changing selector instead of clicking the submit button

mattcoleanderson committed:
add email notfications for data analyst when report is available for their STT

mattcoleanderson committed:
add feedback report alert to tanf page

mattcoleanderson committed:
migrate FeedbackReportAlert to the FeedbackReports component

mattcoleanderson committed:
use JavaScript default params instead of defaultProps

elipe17 commented on tdrs-frontend/src/components/Reports/tdr/TanfSspReports.jsx:
```diff
@@ -20,6 +22,44 @@ const TanfSspReports = ({ stt, isRegionalStaff }) => {
+  const [latestFeedbackReportDate, setLatestFeedbackReportDate] = useState(null)
+
+  // Fetch latest feedback report for Data Analysts
+  useEffect(() => {
```
Two questions. First, since this only drives the visibility and content of the feedback alert component, could it be moved inside of it instead? Since it is also wrapped in the provider you should be able to get the context values you use here. Second, it seems wasteful to fetch all feedback reports just to only use the latest one's created_at date for the alert component. Could we add extra query params to specify we only want the latest report?

mattcoleanderson replied:
I agree with both points. I've gone ahead and moved the useEffect to the FeedbackAlert component. I've also added a latest filter to the reports/views.py queryset

elipe17 commented on tdrs-frontend/src/components/FeedbackReports/FeedbackReportAlert.jsx:
```diff
@@ -0,0 +1,35 @@
+  )
+}
+
+FeedbackReportAlert.propTypes = {
```
I think propTypes is deprecated. If so, we should move away from it

mattcoleanderson replied:
You're right, I've removed it in favor of react state

mattcoleanderson committed:
Merge remote-tracking branch 'origin/5390-feedback-reports-admin-ui' into 5417-feeback-reports-stt-ui

elipe17 commented on tdrs-frontend/src/components/FeedbackReports/STTFeedbackReports.jsx:
```diff
@@ -0,0 +1,226 @@
+
+    try {
+      const response = await axiosInstance.get(
+        `${process.env.REACT_APP_BACKEND_URL}/reports/`,
```
Why no filter by year on this?

mattcoleanderson replied:
Oh yeah, that's a good catch. Fixed.

elipe17 commented on tdrs-frontend/src/components/FeedbackReports/STTFeedbackReports.jsx:
```diff
@@ -0,0 +1,226 @@
+   * Filter reports by applied fiscal year (only updates on Search click)
+   */
+  const filteredReports = reports.filter(
+    (report) => report.year === parseInt(appliedYear, 10)
```
If we hit the backend no matter what each time the user clicks search this filtering should be handled by the backend.

jtimpe replied:
agree with this

mattcoleanderson replied:
Totally agree. Fixed

elipe17 commented on tdrs-frontend/src/components/FeedbackReports/STTFeedbackReports.jsx:
```diff
@@ -0,0 +1,226 @@
+                onClick={handleSearch}
+                disabled={loading}
+              >
+                Search
```
Im sure the UX mockup called for a search button. But should we ask about removing it to keep consistent with the datafile pages now?

mattcoleanderson replied:
@reitermb @victoriaatraft what do y'all think about this? The mockup does show a search button, but it does make sense to remove it for consistency.

@elipe17 I've gone ahead and removed it while we await a response from UX. If it's wanted I can revert the commit.

victoriaatraft replied:
agree to remove for consistency! I think the mockups were made before the auto search feature was implemented. @mattcoleanderson

elipe17 commented on tdrs-frontend/src/components/FeedbackReports/STTFeedbackReportsTable.jsx:
```diff
@@ -0,0 +1,129 @@
+      )
+
+      // Create a blob URL and trigger download
+      const url = window.URL.createObjectURL(new Blob([response.data]))
```
I know we use this logic in other places too. Is this already wrapped into a re-useable function? if not, could we?

mattcoleanderson replied:
There is not a re-usable function for this logic. I've gone made this reusable under the utils dir

mattcoleanderson committed:
fix failing test

mattcoleanderson committed:
Merge branch '5390-feedback-reports-admin-ui' into 5417-feeback-reports-stt-ui

mattcoleanderson committed:
Merge branch '5390-feedback-reports-admin-ui' into 5417-feeback-reports-stt-ui

mattcoleanderson committed:
Merge branch 'develop' into 5417-feeback-reports-stt-ui

jtimpe reviewed (commented):
Looking really good to me so far! A couple things
* I agree with Eric's comments, specifically about doing the `year` filtering on the backend, and removing the search button/using url query params
* Please add testing steps

mattcoleanderson committed:
add new queryset filter for reports view to get the latest report

mattcoleanderson committed:
move the fetchflatestFeedbackReport method from TanfSspReports.jsx to FeedbackReportAlert.jsx. Also removed the proptType usage

mattcoleanderson committed:
Filter by year on api call instead of client side and remove search button

mattcoleanderson committed:
remove PropTypes

mattcoleanderson committed:
extract blob download to a re-useable method

mattcoleanderson committed:
fix failing backend test for reports/views.py

mattcoleanderson committed:
fix failing frontend tests

mattcoleanderson committed:
add JEST_ARG variable to frontend-test task

mattcoleanderson committed:
fix linting errors

mattcoleanderson commented:
> Looking really good to me so far! A couple things
> 
> * I agree with Eric's comments, specifically about doing the `year` filtering on the backend, and removing the search button/using url query params
> * Please add testing steps

@jtimpe  Resolved Eric's comments and added testing steps to the PR

elipe17 reviewed (approved):
two things that arent blocking. 

- are we going to add the url params in this pr or is it written in another ticket?
- is it possible to use the downloadblob func in the other places in the codebase that download files?

mattcoleanderson commented:
> two things that arent blocking. 
> 
> - are we going to add the url params in this pr or is it written in another ticket?
> - is it possible to use the downloadblob func in the other places in the codebase that download files?

@elipe17 

- yes! I forgot we discussed that. I'll add it in this one
- it should be. Just didn't want to scope creep on this ticket. I'll see if doings so is trivial

mattcoleanderson committed:
add bidirectional query params and use queryparams to filter alerts links for centralized feedback

jtimpe reviewed (approved):
Good work!

mattcoleanderson committed:
Merge branch 'develop' into 5417-feeback-reports-stt-ui

mattcoleanderson committed:
Merge branch 'develop' into 5417-feeback-reports-stt-ui

mattcoleanderson committed:
Merge branch 'develop' into 5417-feeback-reports-stt-ui

mattcoleanderson committed:
update feedback reports backend to use date_extracted_on instead of quarter field

mattcoleanderson committed:
update celery task for feedback reports to no longer calculate quarter

mattcoleanderson committed:
update emails for feedback reports to remove quarter field and add date_extracted_on field

mattcoleanderson committed:
update admin feedback report to new design

raftmsohani commented on tdrs-backend/tdpservice/email/templates/feedback/report-available.html:
```diff
@@ -0,0 +1,32 @@
+{% extends 'base.html' %}
+
+{% block content %}
+<p style="color: #000000;">
```
Nesting <p> tag in email clients is not best practice. Email clients do not render paragraph tags correctly. Suggest changing parent <p> to <div>

mattcoleanderson replied:
fixed. changed the outer p tag to a div

raftmsohani reviewed (changes requested):
small nit suggestion

mattcoleanderson committed:
reformat date extracted on from forward slash to hyphen delimitted for passing to backend

mattcoleanderson committed:
make fiscal year a persistent query param for admin upload feedback reports

mattcoleanderson committed:
update STT Feedback Report with changes related to updated design doc

mattcoleanderson committed:
Merge branch 'develop' into 5417-feeback-reports-stt-ui

mattcoleanderson committed:
remove date_extracted_on from read only

mattcoleanderson committed:
update test zip folders and expected zip folder structure

mattcoleanderson committed:
fix linting errors

mattcoleanderson committed:
add coverage to frontend feedback reports tests

mattcoleanderson committed:
fix linting errors

mattcoleanderson committed:
fix padding issue with stt_code

mattcoleanderson committed:
add ability to dismiss alert banner for feedback report and stays dismissed until a new report is available for the given fiscal year

mattcoleanderson committed:
unnest p tags in email template

mattcoleanderson commented:
Based on client feedback, significant UX changes have been implemented. Here's what's new:

Backend Changes

Models & Migrations
  - Replaced quarter field with date_extracted_on (DateField) on both ReportSource and ReportFile models
  - Removed Quarter choices class from both models
  - Updated unique constraint to use date_extracted_on instead of quarter

Celery Tasks
  - Removed all quarter calculation logic (calculate_quarter_from_date, _determine_quarter)
  - Updated zip file structure parsing: now expects FY{YYYY}/R{XX}/F{X}/ format (was YYYY/Region/STT/)
  - Fixed STT code lookup to handle zero-padding (2-digit for states, 3-digit for tribes)

Email Template
  - Updated report-available.html to display date_extracted_on instead of quarter

Frontend Changes

Admin View
  - Added fiscal year selector at top of page
  - Added USWDS date picker for "Data extracted from database on"
  - Content below `<hr>` now hidden until fiscal year is selected
  - Added validation error states for file and date inputs
  - Upload history now filters by selected fiscal year
  - Removed "Fiscal Year" column, added "Data Extracted On" column to history table
  - Fiscal year persists as URL query param (?year=2025)

STT View
  - Fiscal year selector now defaults to no selection (was defaulting to current year)
  - Added placeholder option: "- Select Fiscal Year -"
  - Content below `<hr>` hidden until fiscal year is selected
  - Added H2 header with STT name: "{STT Name} — Fiscal Year {year} — Feedback Reports"
  - Updated table columns: "Generated on", "Reflects data submitted through"
  - Table now shows date_extracted_on instead of quarter

Alert Banner
  - Added dismiss button (X) to alert
  - Dismissal persists in localStorage per fiscal year
  - Alert reappears when a new report is uploaded (different created_at)

mattcoleanderson committed:
Merge branch 'develop' into 5417-feeback-reports-stt-ui

elipe17 reviewed (approved):
Nice job sir

jtimpe reviewed (approved):
Nice work!

ADPennington commented:
@mattcoleanderson I deployed this PR to QASP environment. I'd like to review and test this before it merges.

mattcoleanderson commented:
@ADPennington absolutely! I'm deploying it to A11Y as well so Miles and Victoria can review

mattcoleanderson commented:
@reitermb @victoriaatraft this is ready for UX review again. In dev office hours we had the idea to make the alert banner dismissible. This is the banner shows up on the TANF Data Files page to alert the analyst when a feedback report is available for download. The banner only shows up when a feedback report is available for the analysts STT and the selected fiscal year.

Now there is an x on the right of the banner. If the user clicks it, the banner is dismissed. The dismissal state is saved to the users local storage and keeps the banner dismissed until a new report is uploaded for that STT and fiscal year.

This way, the alert isn't cluttering the analysts screen at all times.

![2026-02-05 12 28 16](https://github.com/user-attachments/assets/6fc062db-b1d8-4cca-8b45-521e4d857aad)

ADPennington commented on tdrs-backend/tdpservice/reports/test/data/README.md:
```diff
@@ -4,50 +4,53 @@ This directory contains test zip files for manually testing the report source wo
+All valid files follow the structure: `FY{YYYY}/R{XX}/F{X}/files`
+
+- **FY{YYYY}**: Fiscal year folder with "FY" prefix (e.g., `FY2025`)
+- **R{XX}**: Region folder with "R" prefix (e.g., `R01`, `R04`)
```
```suggestion
- **RO{XX}**: Region folder with "RO" (i.e. Regional Office) prefix (e.g., `RO01`, `RO04`)
```

@reitermb can you share access to the Sharepoint folder where example zip files are with @mattcoleanderson. Here you'll notice the abovementioned folder structure. Fine to replicate this folder structure for testing purposes, but please dont use the pdfs from Sharepoint in the codebase.

As it stands, without this change, we're getting the result below:

<img width="1241" height="264" alt="Image" src="https://github.com/user-attachments/assets/1d6b0721-abdc-4b74-953b-eff415c92fac" />

mattcoleanderson replied:
@ADPennington What is the folder structure of the contents of your submitted zip file? Because the comments saying `R` instead of `RO`, I'll fix them, but they don't affect the parsing. We just assume that the STT folder, `F{X}` is the 3rd level. So this is the expected structure:

`zip -> FY{YYYY}/RO{XX}/F{X}/files`

Then the 3rd folder is parsed for the STT code.

To get the error that you have it would mean you have 4 nested folders:

`zip -> Folder1/FY{YYYY}/RO{XX}/F{X}/files`

Since `R{XX}` is now the 3rd folder, the parser parses it for the STT code and we get an error.

mattcoleanderson replied:
I see. I just downloaded your zip you uploaded from QASP and the hierarchy was:

`FY2025_07312025.zip -> FY2025_07312025/FY2025/RO1/F9/files`

Where as what the parser expects is:

`FY2025_07312025.zip -> FY2025/RO1/F9/files`

So when unzipped, there is a folder as the same name as the zip folder, followed by another fiscal year folder. I'll update the parser to assume 4 nested folders before the STT folder.

victoriaatraft replied:
@ADPennington I believe this is the [link to the sharepoint files ](https://gorafttech.sharepoint.com/sites/TDRSResearchDesign/Shared%20Documents/Forms/AllItems.aspx?id=%2Fsites%2FTDRSResearchDesign%2FShared%20Documents%2FResearch%20%26%20Design%20%2D%20Secure%2FOFA%2Dgenerated%5FFeedback%5FRpts%2FOneDrive%5F1%5F1%2D23%2D2026%2Ezip&parent=%2Fsites%2FTDRSResearchDesign%2FShared%20Documents%2FResearch%20%26%20Design%20%2D%20Secure%2FOFA%2Dgenerated%5FFeedback%5FRpts)

ADPennington replied:
thank you @mattcoleanderson for clarifying what was happening here. In the meantime, i did remove the top folder so i could test performance with the upload with a typical zip file. it does take about 1-2 minutes to be submitted. probably something to call out for the digit team with any guidance we'll be providing @reitermb

reitermb replied:
Sounds like we're set given an adjustment on the expected folder structure but I wanted to follow up on:

> Here you'll notice the abovementioned folder structure. Fine to replicate this folder structure for testing purposes, but please dont use the pdfs from Sharepoint in the codebase.

Would you mind clarifying what not using the PDF's means here @ADPennington? Is that to say that we're moving away from a couple of the files in a set of feedback reports being PDF based—or something else?

ADPennington replied:
@reitermb i'll ping on MM 😄

mattcoleanderson replied:
@ADPennington fixed. The parser now expects:

`FY2025_07312025.zip -> FY2025_07312025/FY2025/RO1/F9/file`

ADPennington commented on tdrs-backend/tdpservice/reports/test/data/README.md:
```diff
@@ -4,50 +4,53 @@ This directory contains test zip files for manually testing the report source wo
-  └── 4/
-      └── 01/
+FY2025/
+  └── R04/
```
```suggestion
  └── RO4/
```

ADPennington commented on tdrs-backend/tdpservice/reports/test/data/README.md:
```diff
@@ -4,50 +4,53 @@ This directory contains test zip files for manually testing the report source wo
-  └── 4/
-      ├── 01/
+FY2025/
+  └── R04/
```
```suggestion
  └── RO4/
```

ADPennington commented on tdrs-frontend/src/components/FeedbackReports/AdminFeedbackReports.jsx:
```diff
@@ -0,0 +1,396 @@
+  "Choose the date that the data you're uploading was extracted from the database."
+
+/**
+ * AdminFeedbackReports component allows OFA Admins to upload quarterly feedback reports
```
```suggestion
 * AdminFeedbackReports component allows DIGIT Team to upload quarterly feedback reports
```

Can we implement this permissions change here? Only DIGIT Team users will be able to upload these feedback reports.

ADPennington replied:
and sys admin

mattcoleanderson replied:
Should I restrict the upload page to just Digit Team or still allow OFA Admin to see it as well, even though they won't be updating?

mattcoleanderson replied:
Actually better question, what groups should have access to what? There is currently the ability to upload, and the ability to view uploads. 

Right now `OFA Admin` (soon `Digit Team`) can upload and `Data Analyst` can view uploads. Are there any that need to be added or removed from these two permissions?

ADPennington replied:
> Should I restrict the upload page to just Digit Team or still allow OFA Admin to see it as well, even though they won't be updating?

no need for ofa admin. i think we need to revisit its permissions (currently, no one uses this). for this report, digit and sys admin need the ability to upload. and sys admin and data analyst should be able to view uploads. I believe regional staff should be able to view as well but i believe this may be another ticket, hopefully.

mattcoleanderson replied:
Sounds good. I'll have that and the zip parsing fixed and pushed here soon.

ADPennington replied:
oh one more suggestion: digit team should be able to view the uploads as well.

mattcoleanderson replied:
@ADPennington Done. The following permissions should now be reflected:

- Digit Team: upload and view
- OFA Sys Admin: upload and view
- OFA Admin: none
- Data Analyst: view

mattcoleanderson commented:
@reitermb @victoriaatraft Also, I wanted y'all's opinion on a possible change that we discussed in office hours. This doesn't have to be in this PR, we can create a follow up ticket for this.

So the [USWDS Date Picker](https://designsystem.digital.gov/components/date-picker/) that we used for the date extracted on selection, comes with its own quirks that makes it not play very nicely with React. Essentially, the date picker is a vanilla JS component that doesn't integrate with React's state management. The date picker works by hiding the original `<input>` element and UWSDS creates its own input for display. Setting the value is done purely through JS, affectively hiding the change from React, since React state only updates when a native DOM event is triggered.

<img width="501" height="119" alt="image" src="https://github.com/user-attachments/assets/47d96c70-1793-4b17-95d0-c93cab1a45a7" />


So we have to work around this by extracting the value directly from the DOM, which is just not a great solution.

Our recommendation would be to instead use the [USWDS Memorable Date](https://designsystem.digital.gov/components/memorable-date/) because it is purely HTML/CSS. Made up of `<select>` and `<input>` tags. So React state will update when the user interacts with it.

<img width="449" height="156" alt="image" src="https://github.com/user-attachments/assets/756fa472-a663-4e38-bd8a-1bd81bfe99b7" />


We lose the fancy calendar popup UI, but gain proper React integration.

As a side note, USWDS seems to recommend we use the Memorable Date over the Date picker for our use case as well. Under their Guidance section the Memorable Date is recommended for most circumstances, where as the Date Picker is mainly recommended for Scheduling or when knowing the week is necessary.

ADPennington commented:
@mattcoleanderson should we expect to see any logentries to show evidence of the feedback report upload and subsequent email notifications to STTs after a successful upload?

ADPennington commented:
> @mattcoleanderson should we expect to see any logentries to show evidence of the feedback report upload and subsequent email notifications to STTs after a successful upload?

just re-tested and am starting to see logentries :)

ADPennington commented on tdrs-backend/tdpservice/email/templates/feedback/report-available.html:
```diff
@@ -0,0 +1,32 @@
+    </p>
+
+    <p>
+        The report was generated on <b>{{ report_date }}</b>. Please review the feedback and, if needed, resubmit complete and accurate data.
```
```suggestion
        The report was uploaded on <b>{{ report_date }}</b>. Please review the feedback and, if needed, resubmit complete and accurate data.
```

@reitermb @mattcoleanderson recommend using "uploaded" here since this date will align with report_date

mattcoleanderson replied:
@ADPennington done

victoriaatraft commented:
>@reitermb @victoriaatraft this is ready for UX review again. In dev office hours we had the idea to make the alert banner dismissible. This is the banner shows up on the TANF Data Files page to alert the analyst when a feedback report is available for download. The banner only shows up when a feedback report is available for the analysts STT and the selected fiscal year.

Def don't mind dismissing the banner! would just close the gap between the icon and the text lol but im just going off of the gif in the pr, still deploying to a11y. 

> @reitermb @victoriaatraft Also, I wanted y'all's opinion on a possible change that we discussed in office hours. This doesn't have to be in this PR, we can create a follow up ticket for this.
> 
> So the [USWDS Date Picker](https://designsystem.digital.gov/components/date-picker/) that we used for the date
> As a side note, USWDS seems to recommend we use the Memorable Date over the Date picker for our use case as well. Under their Guidance section the Memorable Date is recommended for most circumstances, where as the Date Picker is mainly recommended for Scheduling or when knowing the week is necessary.

Also all for uswds/react integration for components

@mattcoleanderson

reitermb commented:
> @reitermb @victoriaatraft Also, I wanted y'all's opinion on a possible change that we discussed in office hours. This doesn't have to be in this PR, we can create a follow up ticket for this.
> 
> So the [USWDS Date Picker](https://designsystem.digital.gov/components/date-picker/) that we used for the date extracted on selection, comes with its own quirks that makes it not play very nicely with React. Essentially, the date picker is a vanilla JS component that doesn't integrate with React's state management. The date picker works by hiding the original `<input>` element and UWSDS creates its own input for display. Setting the value is done purely through JS, affectively hiding the change from React, since React state only updates when a native DOM event is triggered.
> <img alt="image" width="501" height="119" src="https://private-user-images.githubusercontent.com/49250378/545786404-47d96c70-1793-4b17-95d0-c93cab1a45a7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzAzMjcwMjAsIm5iZiI6MTc3MDMyNjcyMCwicGF0aCI6Ii80OTI1MDM3OC81NDU3ODY0MDQtNDdkOTZjNzAtMTc5My00YjE3LTk1ZDAtYzkzY2FiMWE0NWE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA1VDIxMjUyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5NDMzZDdmNjUxM2E1MzU2YjlkNDI0MjZlZjcxMmE1M2MwYzgwNTc1ZWM1ZmYwODRlZDc2OTU0YTk5OGM5NGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.XqiI4i9T4HJdbrJkvANgXliGLQwc8rnfyEtNtC80JIg">
> 
> So we have to work around this by extracting the value directly from the DOM, which is just not a great solution.
> 
> Our recommendation would be to instead use the [USWDS Memorable Date](https://designsystem.digital.gov/components/memorable-date/) because it is purely HTML/CSS. Made up of `<select>` and `<input>` tags. So React state will update when the user interacts with it.
> <img alt="image" width="449" height="156" src="https://private-user-images.githubusercontent.com/49250378/545781336-756fa472-a663-4e38-bd8a-1bd81bfe99b7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzAzMjcwMjAsIm5iZiI6MTc3MDMyNjcyMCwicGF0aCI6Ii80OTI1MDM3OC81NDU3ODEzMzYtNzU2ZmE0NzItYTY2My00ZTM4LWJkOGEtMWJkODFiZmU5OWI3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA1VDIxMjUyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwZDU3MmU3YzE1NGRmMDA4ZGE1N2MwYWI5ZTk3YjAyOWJjOGNlN2NiNmYwZWVhYjBhNDcxMWFjYmFhNDNkNmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xJkB0unr2buX_iXl8uaD8thk7zQIqhiP7u-HKX7J8yg">
> 
> We lose the fancy calendar popup UI, but gain proper React integration.
> 
> As a side note, USWDS seems to recommend we use the Memorable Date over the Date picker for our use case as well. Under their Guidance section the Memorable Date is recommended for most circumstances, where as the Date Picker is mainly recommended for Scheduling or when knowing the week is necessary.

Works for me to not use the USWDS datepicker but I think I might have a better alternative. Looks like any HTML input with type="date" actually activates a browser-handled HTML/CSS datepicker. Looks like it stores the entered value as yyyy-mm-dd by default, works great even with js fully disabled in Edge, Chrome, and Firefox, and styles well as a USWDS input class. 

Will this play nice with our react @mattcoleanderson?

![firefox_NkM6m60JI1](https://github.com/user-attachments/assets/3971cd22-6d68-4427-93a8-4cf57e01c51e)

```
<label for="event-date" class="usa-label"><strong>Event Date:</strong></label>
 <div class="usa-hint" id="appointment-date-hint">mm/dd/yyyy</div>
<input class="usa-input" type="date" id="event-date" name="event-date">
```

mattcoleanderson committed:
fix expected zip file structure of feedback report source to expect 4 nested folders instead of 3

mattcoleanderson committed:
remove upload permissions for feedback reports from OFA Admin and add them for Digit Team

mattcoleanderson commented:
> > @reitermb @victoriaatraft Also, I wanted y'all's opinion on a possible change that we discussed in office hours. This doesn't have to be in this PR, we can create a follow up ticket for this.
> > So the [USWDS Date Picker](https://designsystem.digital.gov/components/date-picker/) that we used for the date extracted on selection, comes with its own quirks that makes it not play very nicely with React. Essentially, the date picker is a vanilla JS component that doesn't integrate with React's state management. The date picker works by hiding the original `<input>` element and UWSDS creates its own input for display. Setting the value is done purely through JS, affectively hiding the change from React, since React state only updates when a native DOM event is triggered.
> > <img alt="image" width="501" height="119" src="https://private-user-images.githubusercontent.com/49250378/545786404-47d96c70-1793-4b17-95d0-c93cab1a45a7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzAzMjcwMjAsIm5iZiI6MTc3MDMyNjcyMCwicGF0aCI6Ii80OTI1MDM3OC81NDU3ODY0MDQtNDdkOTZjNzAtMTc5My00YjE3LTk1ZDAtYzkzY2FiMWE0NWE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA1VDIxMjUyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5NDMzZDdmNjUxM2E1MzU2YjlkNDI0MjZlZjcxMmE1M2MwYzgwNTc1ZWM1ZmYwODRlZDc2OTU0YTk5OGM5NGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.XqiI4i9T4HJdbrJkvANgXliGLQwc8rnfyEtNtC80JIg">
> > So we have to work around this by extracting the value directly from the DOM, which is just not a great solution.
> > Our recommendation would be to instead use the [USWDS Memorable Date](https://designsystem.digital.gov/components/memorable-date/) because it is purely HTML/CSS. Made up of `<select>` and `<input>` tags. So React state will update when the user interacts with it.
> > <img alt="image" width="449" height="156" src="https://private-user-images.githubusercontent.com/49250378/545781336-756fa472-a663-4e38-bd8a-1bd81bfe99b7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzAzMjcwMjAsIm5iZiI6MTc3MDMyNjcyMCwicGF0aCI6Ii80OTI1MDM3OC81NDU3ODEzMzYtNzU2ZmE0NzItYTY2My00ZTM4LWJkOGEtMWJkODFiZmU5OWI3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA1VDIxMjUyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwZDU3MmU3YzE1NGRmMDA4ZGE1N2MwYWI5ZTk3YjAyOWJjOGNlN2NiNmYwZWVhYjBhNDcxMWFjYmFhNDNkNmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xJkB0unr2buX_iXl8uaD8thk7zQIqhiP7u-HKX7J8yg">
> > We lose the fancy calendar popup UI, but gain proper React integration.
> > As a side note, USWDS seems to recommend we use the Memorable Date over the Date picker for our use case as well. Under their Guidance section the Memorable Date is recommended for most circumstances, where as the Date Picker is mainly recommended for Scheduling or when knowing the week is necessary.
> 
> Works for me to not use the USWDS datepicker but I think I might have a better alternative. Looks like any HTML input with type="date" actually activates a browser-handled HTML/CSS datepicker. Looks like it stores the entered value as yyyy-mm-dd by default, works great even with js fully disabled in Edge, Chrome, and Firefox, and styles well as a USWDS input class.
> 
> Will this play nice with our react @mattcoleanderson?
> 
> ![firefox_NkM6m60JI1](https://private-user-images.githubusercontent.com/28515874/545856971-3971cd22-6d68-4427-93a8-4cf57e01c51e.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzAzMjkzODMsIm5iZiI6MTc3MDMyOTA4MywicGF0aCI6Ii8yODUxNTg3NC81NDU4NTY5NzEtMzk3MWNkMjItNmQ2OC00NDI3LTkzYTgtNGNmNTdlMDFjNTFlLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA1VDIyMDQ0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4NzAxMmFhNGY0Y2I0YWFkN2IwNjM0Yzc1NjgzZjE2YzI4NzBjNTFkMDQzOGJlZTQ5Y2FjMWI2MGVkOGYwM2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jC9D6dH6EttZ2fvpQktXXCxtsGOkQms0Sf8t2aGoJH8)
> 
> ```
> <label for="event-date" class="usa-label"><strong>Event Date:</strong></label>
>  <div class="usa-hint" id="appointment-date-hint">mm/dd/yyyy</div>
> <input class="usa-input" type="date" id="event-date" name="event-date">
> ```

@reitermb I am fine with this approach and it should work fine with react. But are we not supposed to stick with USWDS whenever we can to keep consistent with other gov sites?

mattcoleanderson committed:
change "gernerated" to "uploaded"

mattcoleanderson committed:
Merge branch 'develop' into 5417-feeback-reports-stt-ui

mattcoleanderson commented:
@ADPennington All of your requested changes have been fixed. You can redeploy to QASP and verify when you get a chance. 

Changes made are the following:
- Update expected zip file structure for parser:
   - old: `FY2025_07312025.zip -> FY2025/RO1/F9/files`
   - new: `FY2025_07312025.zip -> FY2025_07312025/FY2025/RO1/F9/files`
- Change which groups have permission to upload feedback reports:
   - OFA Admin: removed permission
   - Digit Team: added permission
   - OFA System Admin: added permission
- In Feedback Report email, changed the word "generated" to "uploaded"

reitermb commented:
> > > @reitermb @victoriaatraft Also, I wanted y'all's opinion on a possible change that we discussed in office hours. This doesn't have to be in this PR, we can create a follow up ticket for this.
> > > So the [USWDS Date Picker](https://designsystem.digital.gov/components/date-picker/) that we used for the date extracted on selection, comes with its own quirks that makes it not play very nicely with React. Essentially, the date picker is a vanilla JS component that doesn't integrate with React's state management. The date picker works by hiding the original `<input>` element and UWSDS creates its own input for display. Setting the value is done purely through JS, affectively hiding the change from React, since React state only updates when a native DOM event is triggered.
> > > <img alt="image" width="501" height="119" src="https://private-user-images.githubusercontent.com/49250378/545786404-47d96c70-1793-4b17-95d0-c93cab1a45a7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzAzMjcwMjAsIm5iZiI6MTc3MDMyNjcyMCwicGF0aCI6Ii80OTI1MDM3OC81NDU3ODY0MDQtNDdkOTZjNzAtMTc5My00YjE3LTk1ZDAtYzkzY2FiMWE0NWE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA1VDIxMjUyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5NDMzZDdmNjUxM2E1MzU2YjlkNDI0MjZlZjcxMmE1M2MwYzgwNTc1ZWM1ZmYwODRlZDc2OTU0YTk5OGM5NGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.XqiI4i9T4HJdbrJkvANgXliGLQwc8rnfyEtNtC80JIg">
> > > So we have to work around this by extracting the value directly from the DOM, which is just not a great solution.
> > > Our recommendation would be to instead use the [USWDS Memorable Date](https://designsystem.digital.gov/components/memorable-date/) because it is purely HTML/CSS. Made up of `<select>` and `<input>` tags. So React state will update when the user interacts with it.
> > > <img alt="image" width="449" height="156" src="https://private-user-images.githubusercontent.com/49250378/545781336-756fa472-a663-4e38-bd8a-1bd81bfe99b7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzAzMjcwMjAsIm5iZiI6MTc3MDMyNjcyMCwicGF0aCI6Ii80OTI1MDM3OC81NDU3ODEzMzYtNzU2ZmE0NzItYTY2My00ZTM4LWJkOGEtMWJkODFiZmU5OWI3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA1VDIxMjUyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwZDU3MmU3YzE1NGRmMDA4ZGE1N2MwYWI5ZTk3YjAyOWJjOGNlN2NiNmYwZWVhYjBhNDcxMWFjYmFhNDNkNmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xJkB0unr2buX_iXl8uaD8thk7zQIqhiP7u-HKX7J8yg">
> > > We lose the fancy calendar popup UI, but gain proper React integration.
> > > As a side note, USWDS seems to recommend we use the Memorable Date over the Date picker for our use case as well. Under their Guidance section the Memorable Date is recommended for most circumstances, where as the Date Picker is mainly recommended for Scheduling or when knowing the week is necessary.
> > 
> > 
> > Works for me to not use the USWDS datepicker but I think I might have a better alternative. Looks like any HTML input with type="date" actually activates a browser-handled HTML/CSS datepicker. Looks like it stores the entered value as yyyy-mm-dd by default, works great even with js fully disabled in Edge, Chrome, and Firefox, and styles well as a USWDS input class.
> > Will this play nice with our react @mattcoleanderson?
> > ![firefox_NkM6m60JI1](https://private-user-images.githubusercontent.com/28515874/545856971-3971cd22-6d68-4427-93a8-4cf57e01c51e.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzAzMjkzODMsIm5iZiI6MTc3MDMyOTA4MywicGF0aCI6Ii8yODUxNTg3NC81NDU4NTY5NzEtMzk3MWNkMjItNmQ2OC00NDI3LTkzYTgtNGNmNTdlMDFjNTFlLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA1VDIyMDQ0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4NzAxMmFhNGY0Y2I0YWFkN2IwNjM0Yzc1NjgzZjE2YzI4NzBjNTFkMDQzOGJlZTQ5Y2FjMWI2MGVkOGYwM2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jC9D6dH6EttZ2fvpQktXXCxtsGOkQms0Sf8t2aGoJH8)
> >       
> >         [
> >           
> >             ![firefox_NkM6m60JI1](https://private-user-images.githubusercontent.com/28515874/545856971-3971cd22-6d68-4427-93a8-4cf57e01c51e.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzAzMjkzODMsIm5iZiI6MTc3MDMyOTA4MywicGF0aCI6Ii8yODUxNTg3NC81NDU4NTY5NzEtMzk3MWNkMjItNmQ2OC00NDI3LTkzYTgtNGNmNTdlMDFjNTFlLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA1VDIyMDQ0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4NzAxMmFhNGY0Y2I0YWFkN2IwNjM0Yzc1NjgzZjE2YzI4NzBjNTFkMDQzOGJlZTQ5Y2FjMWI2MGVkOGYwM2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jC9D6dH6EttZ2fvpQktXXCxtsGOkQms0Sf8t2aGoJH8)
> >           
> >         ](https://private-user-images.githubusercontent.com/28515874/545856971-3971cd22-6d68-4427-93a8-4cf57e01c51e.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzAzMjkzODMsIm5iZiI6MTc3MDMyOTA4MywicGF0aCI6Ii8yODUxNTg3NC81NDU4NTY5NzEtMzk3MWNkMjItNmQ2OC00NDI3LTkzYTgtNGNmNTdlMDFjNTFlLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA1VDIyMDQ0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4NzAxMmFhNGY0Y2I0YWFkN2IwNjM0Yzc1NjgzZjE2YzI4NzBjNTFkMDQzOGJlZTQ5Y2FjMWI2MGVkOGYwM2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jC9D6dH6EttZ2fvpQktXXCxtsGOkQms0Sf8t2aGoJH8)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://private-user-images.githubusercontent.com/28515874/545856971-3971cd22-6d68-4427-93a8-4cf57e01c51e.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzAzMjkzODMsIm5iZiI6MTc3MDMyOTA4MywicGF0aCI6Ii8yODUxNTg3NC81NDU4NTY5NzEtMzk3MWNkMjItNmQ2OC00NDI3LTkzYTgtNGNmNTdlMDFjNTFlLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA1VDIyMDQ0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4NzAxMmFhNGY0Y2I0YWFkN2IwNjM0Yzc1NjgzZjE2YzI4NzBjNTFkMDQzOGJlZTQ5Y2FjMWI2MGVkOGYwM2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jC9D6dH6EttZ2fvpQktXXCxtsGOkQms0Sf8t2aGoJH8)
> >         
> >       
> > ```
> > <label for="event-date" class="usa-label"><strong>Event Date:</strong></label>
> >  <div class="usa-hint" id="appointment-date-hint">mm/dd/yyyy</div>
> > <input class="usa-input" type="date" id="event-date" name="event-date">
> > ```
> 
> @reitermb I am fine with this approach and it should work fine with react. But are we not supposed to stick with USWDS whenever we can to keep consistent with other gov sites?

I think this is a situation where it's a question of the input itself being USWDS-themed and the datepicker being a graceful enhancement on top of it that keeps the added ease of entry without creating a situation where anybody's going to look at the datepicker and clock that it's styled a little differently (like they might if for instance we just had a stock HTML input without USWDS styling on it hanging out in a form with USWDS inputs). The difference here is that the uniqueness of the datepicker overlay doesn't really invite that comparison if it's not placed side by side with the USWDS datepicker.

I might play around with what it would take to make a pure HTML/CSS version of the USWDS datepicker though. We've submitted enhancements in the past to USWDS components—namely a fix for the combobox early on in the project.

mattcoleanderson commented:
> it does take about 1-2 minutes to be submitted. probably something to call out for the digit team with any guidance we'll be providing @reitermb

@ADPennington do you mean after hitting submit and waiting for the upload, or after it has been uploaded and the record says pending while being parsed?

If the latter then how is your internet speed? I just tested this in the QASP env by submitting the file with actual data in it that you uploaded (I removed the 4th folder since I haven't deployed the new changes yet), and it ran rather quickly. The zipped file size was 40 MB. Took about 7 seconds to submit to S3 and 5 seconds to parse.

ADPennington commented:
> > it does take about 1-2 minutes to be submitted. probably something to call out for the digit team with any guidance we'll be providing @reitermb
> 
> @ADPennington do you mean after hitting submit and waiting for the upload, or after it has been uploaded and the record says pending while being parsed?
> 
> If the latter then how is your internet speed? I just tested this in the QASP env by submitting the file with actual data in it that you uploaded (I removed the 4th folder since I haven't deployed the new changes yet), and it ran rather quickly. The zipped file size was 40 MB. Took about 7 seconds to submit to S3 and 5 seconds to parse.

@mattcoleanderson it was the former -- after clicking submit, it's taking some time to see evidence on the screen that it was submitted. 
<img width="859" height="573" alt="Screenshot 2026-02-06 082436" src="https://github.com/user-attachments/assets/28cb32e9-6e13-40d5-aeb1-9c901097bd04" />

mattcoleanderson committed:
fix backend tests

mattcoleanderson commented:
> > > it does take about 1-2 minutes to be submitted. probably something to call out for the digit team with any guidance we'll be providing @reitermb
> > 
> > 
> > @ADPennington do you mean after hitting submit and waiting for the upload, or after it has been uploaded and the record says pending while being parsed?
> > If the latter then how is your internet speed? I just tested this in the QASP env by submitting the file with actual data in it that you uploaded (I removed the 4th folder since I haven't deployed the new changes yet), and it ran rather quickly. The zipped file size was 40 MB. Took about 7 seconds to submit to S3 and 5 seconds to parse.
> 
> @mattcoleanderson it was the former -- after clicking submit, it's taking some time to see evidence on the screen that it was submitted. <img alt="Screenshot 2026-02-06 082436" width="859" height="573" src="https://private-user-images.githubusercontent.com/63075587/546209737-28cb32e9-6e13-40d5-aeb1-9c901097bd04.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzA0MDEyMjUsIm5iZiI6MTc3MDQwMDkyNSwicGF0aCI6Ii82MzA3NTU4Ny81NDYyMDk3MzctMjhjYjMyZTktNmUxMy00MGQ1LWFlYjEtOWM5MDEwOTdiZDA0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA2VDE4MDIwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUwZTlhYWFjN2JjMjc4OTdkMjM4NWE2MjI5MjRmOWZhNjQ0ZThkMDY0MWVhYjdmODkxNzkzOTkxY2EwMGNhZWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.GVoEiqnN7QsdMJ-jE8MYQCrsYFaoqoeozXL42WHl_Wo">

@ADPennington  Interesting, I'm getting a much quicker time on the QASP env than that

<img width="737" height="589" alt="image" src="https://github.com/user-attachments/assets/efdef3ba-12cd-406f-ac25-725a141d458f" />

reitermb commented:
@mattcoleanderson could we easily swing a spinner & a please remain on page message here just so there's some immediate feedback that things are happening? Good spin-off ticket territory too if that's better.

mattcoleanderson commented:
> @mattcoleanderson could we easily swing a spinner & a please remain on page message here just so there's some immediate feedback that things are happening? Good spin-off ticket territory too if that's better.

@reitermb It's doable. We currently have a a state change of the upload button while uploading, but it isn't animated like an actual spinner would be.

<img width="189" height="116" alt="image" src="https://github.com/user-attachments/assets/5ecf6f4f-bf48-4ec7-bd33-f90d6ed28722" />

I would recommend doing it on another ticket though if we can go without for now, so we can get this one moving along.

mattcoleanderson committed:
Merge branch 'develop' into 5417-feeback-reports-stt-ui

mattcoleanderson committed:
align the banner text to the left

reitermb commented:
Just capturing two small issues that would be great to fix before merging (though I don't think they're severe enough to block it if we prioritize follow-on fixes)
![firefox_elQ3VokOGT](https://github.com/user-attachments/assets/1799bdb0-784a-4f9c-b0b5-b5c99cbdf4d2)
Bug with the datepicker where entering an error state breaks the picker until a page refresh

Also noticing that the success banner doesn't receive focus / get passed to the screenreader with an aria-alert (set to polite) when it appears but I believe that's tied to a broader a11y issue that's emerged that I still owe a ticket on as the success banner on TANF pages also isn't reading out. If there's an obvious quick fix on this that's awesome but I'll assume it needs its own ticket if not.

mattcoleanderson committed:
fixed bug where datepicker is unresponsive after entering an error state

mattcoleanderson commented:
> Just capturing two small issues that would be great to fix before merging (though I don't think they're severe enough to block it if we prioritize follow-on fixes) ![firefox_elQ3VokOGT](https://private-user-images.githubusercontent.com/28515874/546456759-1799bdb0-784a-4f9c-b0b5-b5c99cbdf4d2.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzA2NTUyNzEsIm5iZiI6MTc3MDY1NDk3MSwicGF0aCI6Ii8yODUxNTg3NC81NDY0NTY3NTktMTc5OWJkYjAtNzg0YS00ZjljLWIwYjUtYjVjOTljYmRmNGQyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA5VDE2MzYxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliZmMzNmRhMDQ5MmZmMmY0OTEzNmRjM2ZjNDBlZjAxMzM1ZDU3OTJiOWM2YjVkMTA2MzI0YTE0NDZiZjU1MjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.iurmL5b-lHUhNPdvlKjnJXOfFp6udJurkFCMLobKPYE) [ ![firefox_elQ3VokOGT](https://private-user-images.githubusercontent.com/28515874/546456759-1799bdb0-784a-4f9c-b0b5-b5c99cbdf4d2.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzA2NTUyNzEsIm5iZiI6MTc3MDY1NDk3MSwicGF0aCI6Ii8yODUxNTg3NC81NDY0NTY3NTktMTc5OWJkYjAtNzg0YS00ZjljLWIwYjUtYjVjOTljYmRmNGQyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA5VDE2MzYxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliZmMzNmRhMDQ5MmZmMmY0OTEzNmRjM2ZjNDBlZjAxMzM1ZDU3OTJiOWM2YjVkMTA2MzI0YTE0NDZiZjU1MjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.iurmL5b-lHUhNPdvlKjnJXOfFp6udJurkFCMLobKPYE) ](https://private-user-images.githubusercontent.com/28515874/546456759-1799bdb0-784a-4f9c-b0b5-b5c99cbdf4d2.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzA2NTUyNzEsIm5iZiI6MTc3MDY1NDk3MSwicGF0aCI6Ii8yODUxNTg3NC81NDY0NTY3NTktMTc5OWJkYjAtNzg0YS00ZjljLWIwYjUtYjVjOTljYmRmNGQyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA5VDE2MzYxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliZmMzNmRhMDQ5MmZmMmY0OTEzNmRjM2ZjNDBlZjAxMzM1ZDU3OTJiOWM2YjVkMTA2MzI0YTE0NDZiZjU1MjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.iurmL5b-lHUhNPdvlKjnJXOfFp6udJurkFCMLobKPYE) [ ](https://private-user-images.githubusercontent.com/28515874/546456759-1799bdb0-784a-4f9c-b0b5-b5c99cbdf4d2.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzA2NTUyNzEsIm5iZiI6MTc3MDY1NDk3MSwicGF0aCI6Ii8yODUxNTg3NC81NDY0NTY3NTktMTc5OWJkYjAtNzg0YS00ZjljLWIwYjUtYjVjOTljYmRmNGQyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA5VDE2MzYxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliZmMzNmRhMDQ5MmZmMmY0OTEzNmRjM2ZjNDBlZjAxMzM1ZDU3OTJiOWM2YjVkMTA2MzI0YTE0NDZiZjU1MjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.iurmL5b-lHUhNPdvlKjnJXOfFp6udJurkFCMLobKPYE) Bug with the datepicker where entering an error state breaks the picker until a page refresh
> 
> Also noticing that the success banner doesn't receive focus / get passed to the screenreader with an aria-alert (set to polite) when it appears but I believe that's tied to a broader a11y issue that's emerged that I still owe a ticket on as the success banner on TANF pages also isn't reading out. If there's an obvious quick fix on this that's awesome but I'll assume it needs its own ticket if not.

@reitermb the datepicker issue should be resolved now!

As for the second one, I don't see an immediate fix. I'd say a follow on ticket would be best.

victoriaatraft reviewed (approved):
aside from what @reitermb pointed out, LGTM!
we'll make note of the spin off tickets

mattcoleanderson committed:
add report view and update permissions to ofa sys admin

mattcoleanderson committed:
move ofa sys admin permission addition to seperate migration

ADPennington reviewed (approved):
this looks great @mattcoleanderson 🚀 

@reitermb @victoriaatraft please note that i think Yun will be adding other types of feedback beyond WPR and TL types of reports. We're fine with the language for now, but please explore what sort of updated language she's interested in when meeting to discuss the KC content for the feedback reports. 

<img width="1247" height="491" alt="Image" src="https://github.com/user-attachments/assets/4807ea60-7f90-45f0-bcea-db1ef8da6c87" />

mattcoleanderson committed:
Merge branch 'develop' into 5417-feeback-reports-stt-ui
